### DISCUSSION TITLE
remote webdriver - how?

### DISCUSSION THREAD

**Discussion by dedepene** [seleniumbase/SeleniumBase#3057] [Q&A] [answered]
Hello,

with regular selenium webdriver I can call it remotely on a selenium grid like so:

`driver = webdriver.Remote(command_executor=sel_driver_url, options=options)`

Is there a way to do instantiate a remote driver using SeleniumBase in uc=True mode?

Thanks!!!

**Comment by mdmintz** [comment, marked as answer]
As mentioned in https://github.com/seleniumbase/SeleniumBase/issues/2675#issuecomment-2047434434, "UC Mode is not compatible with the Grid / Remote WebDriver".

There's lots of documentation on using SeleniumBase with a Selenium Grid / Remote WebDriver: ([SeleniumBase/seleniumbase/utilities/selenium_grid/ReadMe.md](https://github.com/seleniumbase/SeleniumBase/blob/master/seleniumbase/utilities/selenium_grid/ReadMe.md) and [SeleniumBase/examples/capabilities](https://github.com/seleniumbase/SeleniumBase/tree/master/examples/capabilities))
but **you can't use UC Mode with the Grid**.

The original undetected-chromedriver doesn't support the Grid / Remote WebDriver either:
https://github.com/search?q=repo%3Aultrafunkamsterdam%2Fundetected-chromedriver+grid&type=issues

The Grid / Remote WebDriver is a problem for UC Mode because while the Grid controls chromedriver, the Grid does not have the root control that is needed. UC Mode requires raw control for modifying a local chromedriver (renamed to uc_driver), as well as for disconnecting the driver from Chrome, and reconnecting to it a short time later. It's also not possible to control `pyautogui` remotely from the Grid. (Note that `pyautogui` is required for multiple UC Mode actions.)
